### DISCUSSION TITLE
Per-file font size enhancements

### DISCUSSION THREAD

**Discussion by tmancini85** [gsantner/markor#1594] [Ideas]
Hi, thanks again for Markor, it's a very powerful tool.

Just few hints after playing around with _File settings->Font size menu_.

**In Edit mode**:
- at the top of the font sizes list there could be a `Default (global-fonte-size)` element (e.g.: `Default (16)`):
    - it will stick (again) this file font size to _Settings->Edit mode->Font Size_;
    - currently, when you set a custom font size for a file, you cannot make it follow again the default size even if you manually choose the same global value (the only workaround to reset this is to rename the file, even within Markor till #1566 is not fixed).

**In View mode** (premise: these behaviors require per-file View mode font size logic, maybe related to #912):
- at the top of the font sizes list there could be a `Default (global-font-size)` element (like above):
    - _global-font-size_ can be the value specified in:
        -  _Settings->View mode->Font Size_;
        -  _Settings->Edit mode->Font Size_, if -1 is set for _Settings->View mode->Font Size_;
    - the displayed font size will be still adjusted by CSS tweak (which is in fact _global_);
- after it (and following Markor philosophy to rule View mode under Edit mode) there could be a `Same as Edit mode (per-file-size-value)` element:
    - this will connect the View mode font size to the value chosen for this specific file in Edit mode;
    - _per-file-size-value_ can also be the value in _Settings->Edit mode->Font Size_, if this file Edit mode is sticked to it.
    - the displayed font size may not be adjusted by CSS;
- chance to set per-file View mode font size:
    - the displayed font size should not be adjusted by CSS here in my opinion.

**For both Edit and View mode**:
- when opening _File settings->Font size menu_, list should be automatically swiped to the current value;
- at the moment, list is displayed at the top and you have to swipe till you found the current font size.

Hope this is not too long-winded, I tried to be as succinct as possible.

**Comment by gsantner** [comment]
"Per-File" settings reset - it needs a general improvement, so there is a default fallback available at all settings.

Edit mode vs view mode font size - it should be needed to only configure one pair of font and font size, which is used everywhere. I don't want to split that up.

**Reply by tmancini85** [reply]
> "Per-File" settings reset - it needs a general improvement, so there is a default fallback available at all settings.

That would be great. But if I choose a custom font size for a specific file, how can I then stick it back to the global default setting? This is why I thought about a `Default` element at the top of the _File settings->Font size list_.

> Edit mode vs view mode font size - it should be needed to only configure one pair of font and font size, which is used everywhere. I don't want to split that up.

I understand, even if it could be useful to have for example large size when editing and then small when viewing (my use case: plain text tables with monospace font which are unfortunately too large for the screen with a sufficiently large font size for editing).
Anyway, the fact that, when you choose a specific font size for a file, only the edit mode is affected is intended or is due to #912?

**Reply by gsantner** [reply]
> That would be great. But if I choose a custom font size for a specific file, how can I then stick it back to the global default setting? This is why I thought about a Default element at the top of the File settings->Font size list.

We talk about the same thing. I talk about something that allows to reset to default setting. But there needs to be such a option for all per-file settings. not just font-size.

**Reply by tmancini85** [reply]
>I talk about something that allows to reset to default setting. 

Sorry, I thought in your first answer you were referring to my reference to #1566, which is to maintain per-file settings and not to reset them.

About the fact that when you choose a specific font size for a file the displayed View mode font size is not changed accordingly,  is it correct my guess that is something related to #912?

**Reply by gsantner** [reply]
Yes

**Reply by gsantner** [reply]
View mode is getting little love.

As in open source and freetime, people choose self if and what they work on. Currently there are not much people. And those who help here - it happens to be edit mode gets like 50x the attention view mode gets. 

I mainly use it alsoo just for viewing finished stuff, or when I need to export it for somebody. Like I have wiki articles and info laying around, and somebody asks me how to do something... then I just use view mode -> export as PDf :smile:

**Reply by tmancini85** [reply]
> it happens to be edit mode gets like 50x the attention view mode gets.

They don't know what they're missing :grin:

Seriously, maybe it is because it's much simpler viewing than editing, so the second needs more work in order to cover new use cases (and so more bugs arise). In fact, the only thing I can think about that's missing in view mode is exactly the possibility to adjust font size (at least for plain text file, and maybe also for markdown; I read some enhancement requests about editing todo.txt in view mode).

Anyway, thanks a lot for the clarification and for the exchange of opinions. If my non-IT full time job will give me a little bit of free time some day, I hope to find a way to start coding again and to give some help for improving... view mode :smile:

**Comment by harshad1** [comment]
Current per-file settings are:
1. Indent size
2. Font size
3. Wrap state
4. Highlight state
5. Preview / edit state
6. Edit position
7. File format
8. View mode Scroll x and y (not actually used?)
9. Auto format enbled

Easiest option will be to add a restore-to-default button for individual files, and a global restore-to-default for all files.
